dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerTest.java

```diff
@@ -817,4 +817,23 @@ public void testFindLinkageErrors_doesNotCatchNoClassDefFoundError()
     Truth.assertThat(firestoreLinkageReport.getMissingMethodErrors()).isNotEmpty();
     Truth.assertThat(firestoreLinkageReport.getMissingFieldErrors()).isNotEmpty();
   }
+
+  @Test
+  public void testFindLinkageErrors_shouldNotFailOnDuplicatedClass()
+      throws RepositoryException, IOException {
+    // There was an issue (#495) where com.google.api.client.http.apache.ApacheHttpRequest is in
+    // both google-http-client-1.19.0.jar and google-http-client-apache-2.0.0.jar.
+    // LinkageChecker.findLinkageErrors was not handling the case properly.
+    // These two jar files are transitive dependencies of the artifacts below.
+    List<Path> paths =
+        ClassPathBuilder.artifactsToClasspath(
+            ImmutableList.of(
+                new DefaultArtifact("io.grpc:grpc-alts:jar:1.18.0"),
+                new DefaultArtifact("com.google.cloud:google-cloud-nio:jar:0.81.0-alpha")));
+
+    LinkageChecker linkageChecker = LinkageChecker.create(paths, paths);
+
+    // This should not raise an exception
+    linkageChecker.findLinkageErrors();
```

Comment:
What's the return value here? Should this report an error, though not throw an exception?

Reply:
Updated the line to clarify the return value. The return value is a valid `LinkageCheckReport` instance.
(Duplicate class files in class path do not mean a linkage error.)